pig-tests/test/org/partiql/pig/tests/PermuteTransformSmokeTests.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package org.partiql.pig.tests
+
+import com.amazon.ionelement.api.ionInt
+import org.junit.jupiter.api.Test
+import org.partiql.pig.tests.generated.ToyLang
+import org.partiql.pig.tests.generated.ToyLangNameless
+import org.partiql.pig.tests.generated.ToyLangToToyLangNamelessVisitorTransform
+import kotlin.test.assertEquals
+
+class PermuteTransformSmokeTests {
+
+    /**
+     * This is a VisitorTransformToToyLangNameless implementation that transforms variable references from names
+     * to indexes.
+     *
+     * The input is an instance of the [ToyLang] domain model and the result is an instance of the
+     * [ToyLangNameless] domain.
+     */
+    private class VariableResolverToPermutedDomain(
+        val scope: Scope = Scope.Global
+    ) : ToyLangToToyLangNamelessVisitorTransform() {
+
+        /** Any variable reference resides in the top scope is undefined. */
```

Comment:
nit: Any variable reference **that**  resides in the top scope is undefined

Reply:
next push